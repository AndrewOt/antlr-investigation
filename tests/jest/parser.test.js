/* global test */
const parserRunner = require('../../parser-runner');
const test1 = require('../test1');
const test2 = require('../test2');
const test3 = require('../test3');
const test4 = require('../test4');
const test5 = require('../test5');

test('test1', () => {
    expect(parserRunner.runSingleTest(test1)).toMatchSnapshot();
});

test('test2', () => {
    expect(parserRunner.runSingleTest(test2)).toMatchSnapshot();
});

test('test3', () => {
    expect(parserRunner.runSingleTest(test3)).toMatchSnapshot();
});

test('test4', () => {
    expect(parserRunner.runSingleTest(test4)).toMatchSnapshot();
});

// Syntax error
test('test5', () => {
    expect(parserRunner.runSingleTest(test5)).toMatchSnapshot();
});
