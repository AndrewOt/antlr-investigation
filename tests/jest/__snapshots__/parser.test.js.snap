// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test1 1`] = `
Object {
  "ast": Object {
    "children": Array [],
    "type": "json",
    "value": "",
  },
}
`;

exports[`test2 1`] = `
Object {
  "ast": Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "type": "key",
            "value": "\\"hello\\"",
          },
          Object {
            "children": Array [],
            "type": "value",
            "value": "\\"world\\"",
          },
        ],
        "type": "keyValuePair",
        "value": "",
      },
    ],
    "type": "json",
    "value": "",
  },
}
`;

exports[`test3 1`] = `
Object {
  "ast": Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "type": "key",
            "value": "\\"test\\"",
          },
          Object {
            "children": Array [],
            "type": "value",
            "value": "\\"testing\\"",
          },
        ],
        "type": "keyValuePair",
        "value": "",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "type": "key",
            "value": "\\"testNumber\\"",
          },
          Object {
            "children": Array [],
            "type": "value",
            "value": "0",
          },
        ],
        "type": "keyValuePair",
        "value": "",
      },
    ],
    "type": "json",
    "value": "",
  },
}
`;

exports[`test4 1`] = `
Object {
  "ast": Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "type": "key",
            "value": "\\"testArray\\"",
          },
          Object {
            "children": Array [],
            "type": "value",
            "value": "[",
          },
        ],
        "type": "keyValuePair",
        "value": "",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "type": "key",
            "value": "\\"testObject\\"",
          },
          Object {
            "children": Array [],
            "type": "value",
            "value": "{",
          },
        ],
        "type": "keyValuePair",
        "value": "",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "type": "key",
            "value": "\\"testBool\\"",
          },
          Object {
            "children": Array [],
            "type": "value",
            "value": "true",
          },
        ],
        "type": "keyValuePair",
        "value": "",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "type": "key",
            "value": "\\"testBool2\\"",
          },
          Object {
            "children": Array [],
            "type": "value",
            "value": "false",
          },
        ],
        "type": "keyValuePair",
        "value": "",
      },
    ],
    "type": "json",
    "value": "",
  },
}
`;

exports[`test5 1`] = `
Object {
  "ast": Object {
    "children": undefined,
    "type": "json",
    "value": "",
  },
}
`;
